sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1733,8 +1733,8 @@ object PushPredicateThroughNonJoin extends Rule[LogicalPlan] with PredicateHelpe
       // attributes produced by the aggregate operator's child operator.
       val (pushDown, stayUp) = splitConjunctivePredicates(condition).partition { cond =>
         val replaced = replaceAlias(cond, aliasMap)
-        cond.deterministic && !cond.throwable &&
-          cond.references.nonEmpty && replaced.references.subsetOf(aggregate.child.outputSet)
+        cond.deterministic && cond.references.nonEmpty &&
```

Comment:
I thought we wanted to change `cond` to `replaced`, but this reverts https://github.com/apache/spark/commit/62956c92cfc74d7523328d168b6d837938cde763 , why?

Reply:
I think `cond.deterministic` and `replaced.deterministic` are equivalent when `aggregate.aggregateExpressions.forall(_.deterministic)`.

Reply:
Is it possible that `cond` is simpler than `replaced`?

Reply:
I think we need to document these hidden assumptions, otherwise code changes here are hard to review.

If we can push down a filter through Aggregate, it means the filter only references the grouping keys. The Aggregate operator can't reduce grouping keys so the filter won't see any new data after pushing down.

Reply:
PR decription updated.

Reply:
Added.